src/library_wasmfs.js

```diff
@@ -52,6 +52,30 @@ var WasmfsLibrary = {
     createPath: function(parent, path, canRead, canWrite) {
       // Cache file path directory names.
       wasmFS$preloadedDirs.push({parentPath: parent, childName: path});
+    },
+    readFile: function(path, opts) {
+      opts = opts || {};
+      opts.encoding = opts.encoding || 'binary';
+      if (opts.encoding !== 'utf8' && opts.encoding !== 'binary') {
+        throw new Error('Invalid encoding type "' + opts.encoding + '"');
+      }
+      
+      // Obtain file size.
+      var pathName = allocate(intArrayFromString(path), ALLOC_NORMAL);
+      var length = _emscripten_wasmfs_file_size(pathName);
+      
+      // Copy the file into a JS buffer on the heap.
+      var buf = _emscripten_wasmfs_readFiles(pathName);
```

Comment:
Discussed in meeting. Thanks for the idea where we could just return the size in the first part of the buffer to avoid having to have a separate call to obtain the size.